### ISSUE TITLE
Use character classes as word boundaries

### ISSUE THREAD

**Issue by bentley** [lichray/nvi2#34]
In current nvi, ‘w’ is not that useful in Japanese text, because it considers a long sentence written without spaces as a single word.

In the old nvi-m17n, ‘w’ moved along katakana/hiragana/kanji boundaries (so, for instance, “本日は晴天なり” would split into “本日” “は” “晴天” “な”). This is also how Xterm handles word selection; see Xterm’s charclass.c. I’ve been told by a Japanese nvi user that this is behavior he misses from nvi-m17n.

It would be useful to break along character class boundaries like this.

**Comment by lichray** [comment]
Yes.  And by relying on the Unicode information, we can also eliminate the platform differences on locale.  Currently we have some code to get the Unicode codepoint https://github.com/lichray/nvi2/blob/1d22313e3390a9b4a3220f2dc78fc28ff23ae6d5/common/key.c#L285 , but only used to display Unicode escape sequences.
